src/v/ssx/sugar.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+// TODO: submit upstream if this is useful.
+#include "json/types.h"
+
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sstring.hh>
+
+#include <fmt/format.h>
+
+#include <cstddef>
+#include <utility>
+
+namespace ssx {
+
+using named_semaphore_units
+  = seastar::semaphore_units<seastar::named_semaphore_exception_factory>;
+
+template<typename T = ss::sstring>
+ss::named_semaphore make_named_semaphore(size_t count, T&& name) {
```

Comment:
It might be worth just aliasing `ss::named_mutex` (which doesn't exist) to `ssx::mutex` and `ss::named_semaphore` to `ssx::semaphore` here, to encourage always naming the things.